### PR TITLE
Layout: consolidate styles and cleanup class names.

### PR DESCRIPTION
**Pull request by mtias**
Long overdue cleanup of layout and navigation classes, making main.scss slimmer. This should have no effect on display.

Please, test thoroughly to make sure we don't introduce regressions.

cc @aduth @ehg @blowery @folletto @shaunandrews @rickybanister 

Test live: https://calypso.live/?branch=update/consolidate-layout-styles

### CODE REVIEW

File: client/devdocs/style.scss [hunk 1]
```diff
@@ -140,6 +140,31 @@
 	margin-top: 48px;
 }
 
+.design-assets {
+	@include breakpoint( "<660px" ) {
+		padding: 0 6px;
+	}
+}
+
+.design-assets h2,
```
Comment (blowery) [inline client/devdocs/style.scss]: any chance we can scope these better while we're in here? they make styling components that want to use these tags a pain.

Comment (blowery) [inline client/devdocs/style.scss]: actually, no, I think it's the rules higher up that are only scoped to `.devdocs__doc`.

Comment (blowery) [inline client/devdocs/style.scss]: Or I'm crazy and this isn't a problem any more.

Comment (blowery) [inline client/devdocs/style.scss]: Nope, it is. I hacked around it in https://github.com/Automattic/wp-calypso/pull/6458/files

Comment (mtias) [inline client/devdocs/style.scss]: Yes, that change seems fine.

**Comment by folletto** [comment]
I tested navigation clicking sections at both Desktop and Mobile sizes:

1. My Sites sections, switching site
2. Reader showing tags
3. Me
4. Notifications

Tested on Chrome, Firefox, Safari, Edge, IE11. Everything seems working as expected...

However, on both Edge and IE11 the Reader content in "Followed Sites" didn't load. While it seems unrelated to this, however doesn't happen in production as far as I could tell.

File: client/layout/index.jsx [hunk 2]
```diff
@@ -165,7 +165,6 @@ Layout = React.createClass( {
 
 	render: function() {
 		var sectionClass = classnames(
-				'wp',
```
Comment (aduth) [inline client/layout/index.jsx]: `layout/logged-out` still assigns the `.wp` class

File: client/layout/logged-out.jsx [hunk 3]
```diff
@@ -29,15 +29,15 @@ const LayoutLoggedOut = ( {
...
+			<div id="content" className="layout__content">
+				<div id="primary" className="layout__primary">
 					{ primary }
 				</div>
-				<div id="secondary" className="wp-secondary">
+				<div id="secondary" className="layout__secondary">
 					{ secondary }
 				</div>
 			</div>
-			<div id="tertiary" className="wp-overlay fade-background">
```
Comment (aduth) [inline client/layout/logged-out.jsx]: Remove `.fade-background` styles from `stylesheets/shared/_animation.scss`?

Comment (aduth) [inline client/layout/logged-out.jsx]: There's a reference to `.wp-overlay` in `stylesheets/shared/functions/_z-index.scss`

**Comment by rickybanister** [comment]
I didn't see any glaring issues, the stuff mentioned above all makes sense.

File: client/layout/logged-out.jsx (hunk 3 above)
Comment (mtias) [inline client/layout/logged-out.jsx]: The whole _animations.scss block can probably be dropped, I'll look into it.

**Comment by alisterscott** [comment]
It looks like the themes list in devdocs has its styles broken by this change: https://github.com/Automattic/wp-calypso/issues/6585

**Pull request merged**